src/components/ConfirmationButton/ConfirmationButton.tsx

```diff
@@ -35,6 +35,11 @@ export type Props = PropsWithSpread<
      * Whether to show a hint about the SHIFT+Click shortcut to skip the confirmation modal.
      */
     showShiftClickHint?: boolean;
+    /**
+     * A handler that determines whether the confirmation modal should be shown.
+     * If it returns `true`, the modal is shown. If it returns `false`, the modal is not shown.
+     */
+    shouldShowModal?: (event: MouseEvent<HTMLButtonElement>) => boolean;
```

Comment:
```suggestion
    perModalOpenHook?: (event: MouseEvent<HTMLButtonElement>) => boolean;
```
Maybe a better name? The name `shouldShowModal` wrongly implies a boolean type, but it is a callback.

Reply:
Couldn't think of one 😃

These are some names gemini suggested:

checkShouldShowModal, getShouldShowModal, confirmationModalPredicate, modalVisibilityCheck, determineShouldShowModal, evaluateShouldShowModal, getShouldShowModal

Reply:
Done, thank you!

Reply:
Wow, how fast are you? That was quick :)

Reply:
I try :D

Could you update the labels please?